clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -79,29 +81,40 @@ private object ApiClient {
   }
 }
 
-// Only cached instances of ApiClient can be constructed.  The actual
-// constructor is private.
-class ApiClient private (
-    apiUrl: String,
-    accessKey: String,
-    secretKey: String,
-    connectionTimeoutSec: String = "",
-    readTimeoutSec: String = ""
+case class APIConfigurations(
+    val apiUrl: String,
+    val accessKey: String,
+    val secretKey: String,
+    val connectionTimeoutSec: String = "",
+    val readTimeoutSec: String = ""
 ) {
   val FROM_SEC_TO_MILLISEC = 1000
 
-  val client = new api.ApiClient
-  client.setUsername(accessKey)
-  client.setPassword(secretKey)
-  client.setBasePath(apiUrl.stripSuffix("/"))
+  var connectionTimeoutMillisec: Int = TIMEOUT_NOT_SET
```

Comment:
More Scala-ish (functional, immutable):

```suggestion
  val connectionTimeoutMillisec: Int =
    if (connectionTimeoutSec != null && !connectionTimeoutSec.isEmpty)
      connectionTimeoutSec.toInt * FROM_SEC_TO_MILLISEC
    else
      TIMEOUT_NOT_SET
```
and the same for readTimeoutSec.

Since it repeats twice, you might

```scala
def stringAsMillisec(s: String): Int {
  return if (s != null && s != "") s.toInt * FROM_SEC_TO_MILLISEC else TIMEOUT_NOT_SET
}
```

and use that.

Reply:
done * 2